File: Exercism Codes/Java/Leap.java
class Leap {

    boolean isLeapYear(int year) {
        boolean leap = false;
    
        if (year % 4 == 0) {
    
          if (year % 100 == 0) {
    
            if (year % 400 == 0)
              leap = true;
            else
              leap = false;
          }
          
          else
            leap = true;
        }
        
        else
          leap = false;

        return leap;
    }

}

File: Exercism Codes/C#/Leap.cs
using System;

public static class Leap
{
    public static bool IsLeapYear(int year)
    {
        if (year % 400 == 0 && year % 100 == 0) return true;
        else if (year % 100 != 0 && year % 4 == 0) return true;
        else return false;
    }
}